### PR TITLE
Caching `LastCommit`

### PR DESCRIPTION
**Pull request by colibrishin**
## Changes
- `LastCommit` needs to be stored to re-join consensus and propose a next height and round block.
- Added four store operations for `LastCommit` : `Get`, `Put`, `Delete`, `GetLastCommitIndices`
- `Vote.Signature` is no longer nullable for reducing complication (`null`, `Empty`, `default`.) Use `ImmutableArray<byte>.Empty` instead.
- LastCommit cache will be cleared in `LastCommitClearThreshold` except for the `LastCommit` of current working `Height`. The Threshold is the total cache count of `LastCommit`.

### CODE REVIEW

File: Libplanet.RocksDBStore/RocksDBStore.cs [hunk 1]
```diff
@@ -1212,6 +1220,97 @@ public override void PruneOutdatedChains(bool noopWithoutCanon = false)
...
+        }
+
+        public override void PutLastCommit(BlockCommit lastCommit)
+        {
+            byte[] key = LastCommitKey(lastCommit.Height);
+
+            if (!(_lastCommitDb.Get(key) is null))
+            {
+                DeleteLastCommit(lastCommit.Height);
+            }
```
Comment (greymistcube) [inline Libplanet.RocksDBStore/RocksDBStore.cs:1257]:
```suggestion
            if (_lastCommitDb.Get(key) is { })
            {
                DeleteLastCommit(lastCommit.Height);
            }
```
Also, is silently overwriting a stored last commit a good idea? 🤔

Comment (colibrishin) [reply Libplanet.RocksDBStore/RocksDBStore.cs]: Oops, I misunderstood that height could be overwritten by round.

File: CHANGES.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ To be released.
  -  Bumped `BlockMetadata.CurrentProtocolVersion` to 4.  [[#PBFT]]
  -  Changed `IPreEvaluationBlockHeader.PreEvaluationHash` type from
     `ImmutableArray<byte>` to `HashDigest<SHA256>`.  [[#PBFT]]
+ -  Added `IStore.GetLastCommit(long height)` method.  [[#PBFT]]
+ -  Added `IStore.PutLastCommit(BlockCommit lastCommit)` method.  [[#PBFT]]
+ -  Added `IStore.DeleteLastCommit(long height)` method.  [[#PBFT]]
+ -  Added `IStore.GetLastCommitIndices()` method.  [[#PBFT]]
```
Comment (limebell) [inline CHANGES.md:38]:
```suggestion
 -  Added `IStore.GetLastCommit(long)` method.  [[#PBFT]]
 -  Added `IStore.PutLastCommit(BlockCommit)` method.  [[#PBFT]]
 -  Added `IStore.DeleteLastCommit(long)` method.  [[#PBFT]]
 -  Added `IStore.GetLastCommitIndices()` method.  [[#PBFT]]
```
We used to skip parameter name in changelog.

Comment (colibrishin) [reply CHANGES.md]: Amended! https://github.com/planetarium/libplanet/pull/2348/commits/4805c8a868eaca9c2224548552f26f9f1d07b180

**Review by limebell — APPROVED** [review]

**Pull request merged**